api/v1alpha1/gatewayclass_types.go

```diff
@@ -116,10 +116,42 @@ type GatewayClassSpec struct {
 }
 
 // GatewayClassParametersObjectReference identifies a parameters object for a
-// gateway class within a known namespace.
-//
-// +k8s:deepcopy-gen=false
-type GatewayClassParametersObjectReference = LocalObjectReference
+// GatewayClass.
+type GatewayClassParametersObjectReference struct {
```

Comment:
Can we preserve the types that `api.TypedLocalObjectReference` uses

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L4414-L4418

Reply:
That's a great question, this was actually based on our [Gateway specific LocalObjectReference type](https://github.com/kubernetes-sigs/service-apis/blob/master/api/v1alpha1/gateway_types.go#L163-L193) that was added by an [earlier PR](https://github.com/kubernetes-sigs/service-apis/pull/73). There's a lot of conversation on that PR covering the rationale for the changes, but unfortunately I don't think we've actually transferred any of that to our docs yet.

Reply:
I agree. In fact, I counter it should be a [`v1.ObjectReference`](https://godoc.org/k8s.io/api/core/v1#ObjectReference) and all the customer object reference types should be removed as they are confusing.

The only place it makes sense is the `LocalObjectReference` since the [core type](https://godoc.org/k8s.io/api/core/v1#LocalObjectReference) lacks the additional fields.